### PR TITLE
Install Trivy with GitHub action

### PR DESCRIPTION
**Pull request by hamistao**
This introduces a new GitHub action for installing Trivy using the deb package as a workaround to properly cache Trivy's vulnerability database.

### CODE REVIEW

**Comment by tomponline** [comment]
needs a rebase please

**Comment by hamistao** [comment]
@tomponline Sure! Also TBD how to proceed on using the github action on every branch for the snap scans. Since the action is only being added to the main branch, works must be done to also use it on the snap scan jobs, since some of them don't checkout to a branch other than main.

**Comment by tomponline** [comment]
> Also TBD how to proceed on using the github action on every branch for the snap scans.

There's no need if the deb works and it supports targeting other branches right?

**Comment by hamistao** [comment]
The problem is just that `./.github/actions/install-trivy` only exists on the main branch, so, for example, when scanning `5.21/stable` we checkout to `stable-5.21` and there is no `./.github/actions/install-trivy` there, so the scan fails. It is a simple fix either way, just have to decide on which approach to take. I am thinking of just downloading the action instead of using it locally from the checked out repo.

**Review by simondeziel — DISMISSED** [review]

**Comment by hamistao** [comment]
@tomponline @simondeziel Now rebased, all green and ready for a review

File: .github/actions/install-trivy/action.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  using: composite
+  steps:
+    # We are using the deb package instead of the trivy action because we need to point to the cache directory
+    # with cache-dir, and there is currently a bug in that functionality on the Trivy GitHub action.
+    # See https://github.com/aquasecurity/trivy-action/issues/12.
+    - name: Install Trivy
+      shell: bash
+      run: |
+        mkdir -p /home/runner/vuln-cache
+        sudo apt-get install wget apt-transport-https gnupg
```
[the remark this answers is not in the document]
Comment (simondeziel) [reply .github/actions/install-trivy/action.yml:14]: Please drop the `apt-transport-https`, my PR to fix trivy's instruction went nowhere due to their CLA :/

[the remark this answers is not in the document]
Comment (hamistao) [reply .github/actions/install-trivy/action.yml]: Oh sorry, I thought we would keep it until they change their docs. Changes made!

**Review by tomponline — APPROVED** [review]

**Pull request merged**